datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -483,13 +614,25 @@ impl CommonSubexprEliminate {
         config: &dyn OptimizerConfig,
     ) -> Result<Transformed<(Vec<Expr>, LogicalPlan)>> {
         let mut expr_stats = ExprStats::new();
-        let arrays = to_arrays(&expr, &mut expr_stats, ExprMask::Normal)?;
-
-        self.rewrite_expr(vec![expr], &[&arrays], input, &expr_stats, config)?
-            .map_data(|(mut new_expr, new_input)| {
+        let (found_common, id_arrays) =
+            self.to_arrays(&expr, &mut expr_stats, ExprMask::Normal)?;
+
+        if found_common {
+            let rewritten = self.rewrite_expr(
+                vec![expr.clone()],
```

Comment:
Extra clone, but the reason is the same as https://github.com/apache/datafusion/pull/10473/files#r1648985272